GSoC-2025/Prisha_Sharma.md

```diff
@@ -0,0 +1,196 @@
+        </tr>
+        <tr>
+            <td><strong>Cellfinder</strong></td>
+            <td><a href="https://github.com/brainglobe/cellfinder/pull/499">#499</a></td>
+            <td>Remove save weights option from training widget (merged)</td>
+        </tr>
+        <tr>
+            <td><strong>Datashuttle</strong></td>
+            <td><a href="https://github.com/neuroinformatics-unit/datashuttle/pull/470">#470</a></td>
+            <td>Attempting to copy in the terminal user interface crashes on headless HPC (merged)</td>
+        </tr>
+    </table>
+</blockquote>
+
+
+
+## Project proposal
+
+_Length: max 1 page_
+
+**Synopsis**
+
+This project aims to enhance CellFinder by adding multi-channel support for brain image classification. The goal is to extend CellFinder’s capabilities to process and classify cells from an arbitrary number of channels, making it more adaptable and improving its usability across different datasets. This enhancement is crucial, as some datasets may contain only a single signal channel, while others may have multiple signal channels carrying valuable information. By improving flexibility and accuracy, this project will make CellFinder a more powerful tool for the open-source neuroscience community.
+
+### Deliverables
+
+- Modify blob detection to support multiple channels.
+
+- Enhance neural network for multi-channel cell classification.
+
+- Ensure comprehensive testing for new functionalities.
+
+- Provide detailed documentation for ease of use.
+
+- Publish a blog post showcasing improvements with 1 & 3-channel images.
+
+-  Bonus (if time allows): Implement a visualization tool to display detected cells across different channels, aiding interpretation.
+
+**Implementation timeline**
+
+<table>
+  <thead>
+    <tr>
+      <th>Timeline</th>
+      <th>Deliverables</th>
+      <th>Hours/Week</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><b>Community Bonding Period</b><br>Focus: Understanding</td>
+      <td>
+        <ol type="1">
+          <li>Discuss project details with mentor and understand the codebase</li>
+          <li>Explore existing multi-channel imaging datasets.</li>
+          <li>Set up the development environment and resolve dependencies</li>
+        </ol>
+      </td>
+      <td>15</td>
+    </tr>
+    <tr>
+      <td><b>Week 1-2</b><br>Focus: Integrating attention into blob detection</td>
+      <td>
+        <ol type="1">
+          <li>Modify the blob detection pipeline to process arbitrary channels through attention.</li>
+          <li>Conduct multi-channel blob detection tests</li>
+          <li>Refine based on mentor's feedback</li>
+        </ol>
+      </td>
+      <td>35</td>
+    </tr>
+    <tr>
+      <td><b>Week 3-4</b><br>Focus: Neural network integration</td>
+      <td>
+        <ol type="1">
+          <li>Integrate attention modules into the neural network classifier to classify cell candidates from brain images with an arbitrary number of channels  .</li>
```

Comment:
I'm not sure attention is appropriate here (but happy for you to convince me otherwise). It should be possible to expand the network to take  N channels as input and re-train.